cluster/gce/gci/configure-helper.sh

```diff
@@ -1453,39 +1453,57 @@ function create-master-etcd-apiserver-auth {
 }
 
 function assemble-docker-flags {
-  echo "Assemble docker command line flags"
-  local docker_opts="-p /var/run/docker.pid --iptables=false --ip-masq=false"
+  echo "Assemble docker options"
+  cat <<EOF >/etc/docker/daemon.json
+{
+  "pidfile": "/var/run/docker.pid",
+  "iptables": false,
+  "ip-masq": false,
+EOF
   if [[ "${TEST_CLUSTER:-}" == "true" ]]; then
-    docker_opts+=" --log-level=debug"
+    cat <<EOF >>/etc/docker/daemon.json
```

Comment:
for DRY, abstract this in a method that can be called from here:

```
if [[ "${XYZ:-}" == "true" ]]; then
  adddockeropt "foo"
fi
```

Reply:
added a method addockeropt